Sources/SwiftParserDiagnostics/ParseDiagnosticsGenerator.swift

```diff
@@ -615,6 +622,30 @@ public class ParseDiagnosticsGenerator: SyntaxAnyVisitor {
     if shouldSkip(node) {
       return .skipChildren
     }
+    // Emit a custom diagnostic for an unexpected '...' after the type name.
+    if node.each?.presence == .present {
+      removeToken(
+        node.unexpectedBetweenNameAndColon,
+        where: { $0.tokenKind == .ellipsis },
+        message: { _ in .typeParameterPackEllipsis }
+      )
+    } else if let unexpected = node.unexpectedBetweenNameAndColon,
+              let unexpectedEllipsis = unexpected.onlyToken(where: { $0.tokenKind == .ellipsis }) {
```

Comment:
Instead of force unwrapping below, I would add `let each = node.each` to the condition.